src/tools/rbd/action/ImageMeta.cc

```diff
@@ -97,9 +97,13 @@ static int do_metadata_set(librbd::Image& image, const char *key,
 static int do_metadata_remove(librbd::Image& image, const char *key)
 {
   int r = image.metadata_remove(key);
-  if (r < 0) {
-    std::cerr << "failed to remove metadata " << key << " of image : "
-              << cpp_strerror(r) << std::endl;
+  if (r < 0 && r != -ENOENT) {
+      std::cerr << "failed to remove metadata " << key << " of image : "
+                << cpp_strerror(r) << std::endl;
+  }
+  else if(r == -ENOENT){
```

Comment:
Nit: move the ```-ENOENT``` check to be the first check so you don't need to exclude it from the other ```if``` conditional. Also, the braces should be on the same line as the ```else``` (i.e. ```} else if (xyz) {```.